src/main/scala/sigmastate/Values.scala

```diff
@@ -424,7 +426,12 @@ object Values {
 
   case class ConcreteCollection[V <: SType](items: IndexedSeq[Value[V]], elementType: V)
     extends EvaluatedCollection[V] {
-    override val opCode: OpCode = ConcreteCollectionCode
+    override val opCode: OpCode =
+      if (elementType == SBoolean
+        && items.forall(i => i.isInstanceOf[Constant[_]] || i.isInstanceOf[BooleanConstant]))
```

Comment:
@ergomorphic what's the difference between `BooleanConstant` and `Constant[SBoolean.type]`? Both might be in ConcreteCollection.

Reply:
@greenhat I thought it will be harder but #179.